roles/dns/tasks/main.yml

```diff
@@ -27,3 +27,7 @@
   with_subelements:
   - "{{ dns_records_add | default({}) }}"
   - entries
+  register: nsupdate_result
+  until: nsupdate_result.dns_rc == 0
```

Comment:
see official docs as to why that is:
http://docs.ansible.com/ansible/latest/nsupdate_module.html#return-values